main/Filters.hpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * @file Filters.hpp
+ * @author your name (you@domain.com)
+ * @brief 
+ * @version 0.1
+ * @date 2024-11-05
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+#pragma once
+
+class Filter {
+public:
+	virtual float operator() (float value) = 0;
+};
+
+template <int av_size = 2>
```

Comment:
Comment template and why 2? 2 looks like  a magic number. You can use a #define for the 2